### DISCUSSION TITLE
https without ssl

### DISCUSSION THREAD

**Discussion by mehmetcolgecen** [nginx/kubernetes-ingress#2895] [Q&A] [answered]
I am working with both http and https ports on different apps. if I use this approach I suppose all ingresses is going to redirected to 443 port. But I need to use both 80 and 443 ports on different hosts without any TLS configuration.

I think "ingress.kubernetes.io/ssl-redirect" annotation should work for this purpose but when I use it I cant track any redirection on conf file. But it should provide redirection according to:

> Sets an unconditional 301 redirect rule for all incoming HTTP traffic to force incoming traffic over HTTPS.

If I use "nginx.org/redirect-to-https" I can track redirection on the conf but it is based on http_x_forwarded_proto and it requires SSL termination but I dont use TLS.

I can see following snippet on /etc/nging/conf.d/xxx.conf when I use nginx.org/redirect-to-https this annotation and If I am not wrong this is not my solution.
```
        if ($http_x_forwarded_proto = 'http') {
                return 301 https://$host$request_uri;
        }
```

But I don't se any snippet on /etc/nging/conf.d/xxx.conf when I use ingress.kubernetes.io/ssl-redirect. I think I should see some snippet something like below which may be a solution of my problem.
`
return 301 https://$host$request_uri;
`

Is there any workaround for my situation or should I wait for the update?

Thank you.

**Comment by brianehlert** [comment, marked as answer]
If the Ingress Controller is not performing SSL termination, and you have SSL traffic - the Ingress Controller is performing SSL pass-through for the SSL traffic.

That sounds like:
- you are forwarding from http to https
- your upstream service is providing SSL

You will need to use TransportServer for both port 80 and 443 then, assuming they are the same hostname.
https://github.com/nginxinc/kubernetes-ingress/tree/main/examples/custom-resources/tls-passthrough

Assuming I am correctly understanding what you are trying to accomplish.

**Reply by brianehlert** [reply]
I neglected to mention.
The annotation `ingress.kubernetes.io/ssl-redirect` is not understood by this implementation.  That annotation is specific to the community ingress controller implementation.

A list of annotations for this project can be found here: https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations#summary-of-annotations
The annotations for this project have the prefix `nginx.org`

**Reply by gloorfindel** [reply]
@brianehlert Can you clarify your comment about ingress.kubernetes.io/ssl-redirect. The link you referenced above does include `ingress.kubernetes.io/ssl-redirect` as a valid annotation for this project.

**Reply by brianehlert** [reply]
You are right.  My mistake.  One of the few exceptions.
